lib/jsonapi/request_parser.rb

```diff
@@ -266,7 +265,10 @@ def parse_filters(filters)
             return @errors.concat(Exceptions::FilterNotAllowed.new(filter_method).errors)
           end
 
-          unless @include_directives.model_includes.include?(relationship.name.to_sym)
+          # Supports only just-included or top-level of multiple-relationship includes, e.g.,
+          # ?include=parent,parent.child&filter[parent.arms]=3
+          filtered_relationship_included = @include_directives.model_includes.include?(relationship.name.to_sym) || @include_directives.model_includes.any? { |include| include.key?(relationship.name.to_sym) }
```

Comment:
This is needed because `@include_directives` turns from an array of included symbols (like `[:parent]`) to an array of objects that point to child-includes like `[{parent: :child}]`.

Reply:
nit: the OR condition added here makes this long and hard-to-grok. would it possibly be cleaner to understand in a named method extract?